### DISCUSSION TITLE
Target date for full Visual Studio support?

### DISCUSSION THREAD

**Discussion by nholland20** [microsoft/DacFx#317] [General]
Building off of this question: https://github.com/microsoft/DacFx/discussions/188. Is there a target date for full Visual Studio support?

Issues that we have been having with the latest release and Visual Studio: 
1. When opened as a project, Visual Studio automatically inserts all of the folders and subfolders into the sqlproj file. (e.g., < Folder Include=<folder_path> />)
2. Visual Studio doesn't automatically pull in the sql files. (We've tried working around it for now with <Build Include="**\*.sql" />)
3. What we haven't been able to work around: With a custom output path, <OutputPath>.\newFolder\</OutputPath>, Visual Studio does not properly ignore .sql files generated during build and publish. If these files are present, Visual Studio will not load the project or solution. The error is: "Length cannot be less than zero. Parameter name: length."
4. We have to manually clean the folder between builds, as outlined here: https://github.com/microsoft/DacFx/issues/180#issuecomment-1374573986

**Comment by ErikEJ** [comment]
VS 2022 17.12 preview has new project type (opt in) .sqlprojx based on Micrsoft.Build.Sql

**Comment by PsyChonek** [comment]
Any news? All the other IDEs have already dropped support in favor of the new SDK, but VS is still holding on. We can't use the preview package without changing our solution structure. Without migration, developers can't fully use VSCode and other IDEs.

**Comment by COF-RNorman** [comment]
So as a note while doing some experimentation...
I am typically very diligent about keeping my NuGet and project references up to date and I have been using Visual Studio 2022 to edit and manage the new SQL SDK style projects. Recently many of my projects stopped loading in Visual Studio 2022. What I discovered is that any reference beyond the 1.0.0 release is breaking the tooling in Visual Studio 2022. Also as previously noted, the 2026 and higher versions don't support it at all (yet).

So unless you are using other tooling (Like Visual Studio Code or SQL Server Management Studio) you must stay at version 1.0.0 until official support is updated. It would be useful to possibly mention this in the release notes until support is updated
https://www.nuget.org/packages/Microsoft.Build.Sql/

This worked in my Visual Studio 2022

> `<Sdk Name="Microsoft.Build.Sql" Version="1.0.0" />`

This failed in Visual Studio 2022 with various SDK errors

> `<Sdk Name="Microsoft.Build.Sql" Version="2.3.0" />`

> C:\Repos\Training\SampleDatabase1\SampleDatabase1.sqlproj : error  : The SDK 'Microsoft.NET.SDK.WorkloadAutoImportPropsLocator' specified could not be found.  C:\Program Files\dotnet\sdk\10.0.401\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.ImportWorkloads.props
> 
> <img width="539" height="227" alt="image" src="https://github.com/user-attachments/assets/8f09979a-f23d-43a4-bfd7-f091abf87491" />

**Reply by COF-RNorman** [reply]
One more note is that I just generated a new project in SQL Server Management studio and can confirm that the version that it uses points to 2.1.0

`<Sdk Name="Microsoft.Build.Sql" Version="2.1.0" />`

SQL Server Management Studio Version Info:
SQL Server Management Studio - 22.10.1+20.12210.168
SQL Server Management Objects (SMO) - 18.100.11.36+8ec380f58be7e8766b7fafa2b61ec16ac0dee5a4
Microsoft T-SQL Parser - 18.0.7.0+88067086123bf2a0756082eee9e83f6386d2d360.88067086123bf2a0756082eee9e83f6386d2d360
Microsoft Analysis Services Client Tools - 22.1.17.0
Microsoft Data SqlClient (MDS) - 6.1.5+5bcf2003f4bab96e32aff1e2158393f9f4fcd1ed
Microsoft SQL Server Data-Tier Application Framework (DacFX) - 170.4.83.3+6b5a249e797735a88101d9bd17c72ef3fd0a0a87
Microsoft SQL Copilot Server - 1.0.127.0
Microsoft .NET Framework - 4.0.30319.42000
Operating System - 10.0.26200

**Reply by COF-RNorman** [reply]
I did not include my Visual Studio info for reference (Only Included Sql Server Management Studio info).

I opened a bug report for the above observation, however as of now it was closed without action.
#848 

> I should have shared the version information from Visual Studio. Attaching that below:
> 
> Microsoft Visual Studio Professional 2022
> Version 17.14.41 (September 2026)
> VisualStudio.17.Release/17.14.41+37710.0.-september.2026-
> Microsoft .NET Framework
> Version 4.8.09221
> 
> Installed Version: Professional
> 
> Azure App Service Tools v3.0.0 17.14.156.658
> Azure App Service Tools v3.0.0
> 
> C# Tools 4.14.0-3.26424.7+a528c90a2c019780d2fda6fd2038ba39d7a2abd4
> C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.
> 
> Common Azure Tools 1.10
> Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.
> 
> NuGet Package Manager 6.14.3
> NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/
> 
> SQL Server Data Tools SDK Style (Preview) 17.14.13.1
> Microsoft SQL Server Data Tools for SDK Projects
> 
> Visual Basic Tools 4.14.0-3.26424.7+a528c90a2c019780d2fda6fd2038ba39d7a2abd4
> Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.
